src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -50,14 +52,18 @@
  */
 public class GameRunner {
 
+  public enum GameMode { SWING_CLIENT, HEADLESS_BOT }
```

Comment:
Important enum, the headless server code was copy pasted here and then put behind this enum. After that we simplify.